crates/forge_loader/src/manifest.rs

```diff
@@ -59,23 +81,159 @@ enum Interval {
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
 struct ScheduledTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
     interval: Interval,
 }
 
+// Maps to Web Trigger under Common Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct RawTrigger<'a> {
+    key: &'a str,
+    function: &'a str,
+}
+
+// maps to Trigger under Common Modules
 #[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
-struct WorkflowPostFunction<'a> {
+struct EventTrigger<'a> {
     #[serde(flatten, borrow)]
     raw: RawTrigger<'a>,
+    #[serde(borrow)]
+    events: Vec<&'a str>,
 }
 
+// Compass Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CompassAdminPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct ComponentPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct DataProvider<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    callback: JustFunc<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CompassGlobalPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct TeamPage<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+
+// Confluence Modules
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct ContentAction<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+}
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct ContentByLineItem<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    #[serde(borrow)]
+    dynamic_properties: JustFunc<'a>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+struct MacroMod<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    config: JustFunc<'a>,
+    export: JustFunc<'a>,
+}
+
+// Jira Modules
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct JiraAdminPage<'a> {
+    key: &'a str,
+    function: &'a str,
+    title: &'a str,
+    resolver: Resolver<'a>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CustomField<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    value: Option<&'a str>,
+    search_suggestions: Option<&'a str>,
+    edit: Option<&'a str>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct CustomFieldType<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    value: Option<&'a str>,
+    edit: Option<&'a str>,
+    context_config: Option<&'a str>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct DashboardGadget<'a> {
+    #[serde(flatten, borrow)]
+    common_keys: CommonKey<'a>,
+    edit: Option<&'a str>,
```

Comment:
```suggestion
    edit: Option<JustFunc<'a>>,
```
The edit property is a [struct](https://developer.atlassian.com/platform/forge/manifest-reference/modules/jira-dashboard-gadget/#properties) that may have a function property.